packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -317,13 +317,16 @@ class Auth {
         const refreshToken = refreshTokenRegistered || refreshTokenGuest
         if (refreshToken) {
             try {
-                return this.queueRequest(
+                return await this.queueRequest(
                     () => helpers.refreshAccessToken(this.client, {refreshToken}),
                     !!refreshTokenGuest
                 )
-            } catch {
-                // If anything bad happens during refresh token flow
-                // we continue with the PKCE guest user flow.
+            } catch (error) {
+                // If the refresh token is invalid, we need to re-login the user
+                if (error instanceof Error && error.message === '400 Bad Request') {
```

Comment:
nit, maybe check the error status code is 400, or even better to check the error message is `invalid refresh_token`.

(I believe the error message `400 Bad Request` comes from isomorphic lib, if that changes in the future, this code will break)

Reply:
I believe the response is attached to the error! so you could do this

```
const json = await error.response.json()
console.log(json.message) // invalid refresh_token
```